src/terms/exact_exchange.jl

```diff
@@ -0,0 +1,76 @@
+@doc raw"""
+Exact exchange term: the Hartree-Exact exchange energy of the orbitals
+```math
+-1/2 ∑_{nm} f_n f_m ∫∫ \frac{ψ_n^*(r)ψ_m^*(r')ψ_n(r')ψ_m(r)}{|r - r'|} dr dr'
+```
+"""
+struct ExactExchange
+    scaling_factor::Real  # to scale the term
+end
+ExactExchange(; scaling_factor=1) = ExactExchange(scaling_factor)
+(exchange::ExactExchange)(basis) = TermExactExchange(basis, exchange.scaling_factor)
+function Base.show(io::IO, exchange::ExactExchange)
+    fac = isone(exchange.scaling_factor) ? "" : "scaling_factor=$(exchange.scaling_factor)"
+    print(io, "ExactExchange($fac)")
+end
+struct TermExactExchange <: Term
+    scaling_factor::Real  # scaling factor, absorbed into poisson_green_coeffs
+    poisson_green_coeffs::AbstractArray
+end
+function TermExactExchange(basis::PlaneWaveBasis{T}, scaling_factor) where T
+    poisson_green_coeffs = 4T(π) ./ [sum(abs2, basis.model.recip_lattice * G)
+                                     for G in to_cpu(G_vectors(basis))]
+    poisson_green_coeffs[1] = 0
+    TermExactExchange(T(scaling_factor), scaling_factor .* poisson_green_coeffs)
+end
+
+# Note: Implementing exact exchange in a scalable and numerically stable way, such that it
+# rapidly converges with k-points is tricky. This implementation here is far too simple and
+# slow to be useful.
+#
+# ABINIT/src/66_wfs/m_getghc.F90
+# ABINIT/src/66_wfs/m_fock_getghc.F90
+# ABINIT/src/66_wfs/m_prep_kgb.F90
+# ABINIT/src/66_wfs/m_bandfft_kpt.F90
```

Comment:
I'd rather not give explicit references to code (as that might give the impression that we reproduced the implementation instead of a clean-room reimplementation)